client/txn.go

```diff
@@ -288,12 +307,22 @@ func (txn *Txn) CommitInBatch(b *Batch) error {
 
 // Commit sends an EndTransactionRequest with Commit=true.
 func (txn *Txn) Commit() error {
-	return txn.sendEndTxnCall(true /* commit */)
+	if txn.txn.Writing {
+		return txn.sendEndTxnCall(true /* commit */)
+	}
+	return nil
 }
 
 // Rollback sends an EndTransactionRequest with Commit=false.
 func (txn *Txn) Rollback() error {
-	return txn.sendEndTxnCall(false /* commit */)
+	var err error
+	if txn.txn.Writing {
+		err = txn.sendEndTxnCall(false /* commit */)
+	}
+	// Explicitly set the status as ABORTED so that higher layers
+	// know that this transaction has ended.
+	txn.txn.Status = proto.ABORTED
```

Comment:
I'm dissatisfied with needing to do this. The other alternatives are adding a new state or another field to the protobuf.